src/mono/configure.ac

```diff
@@ -3921,6 +3921,14 @@ if test x$host_win32 = xno; then
 
 	AC_CHECK_DECL(IN_EXCL_UNLINK, [AC_DEFINE(HAVE_IN_EXCL_UNLINK, 1, [IN_EXCL_UNLINK])], [], [[#include <sys/inotify.h>]])
 
+	ORIG_CFLAGS="$CFLAGS"
+	CFLAGS="$CFLAGS -Werror=strict-prototypes"
+
+	AC_CHECK_FUNCS(madvise)
+	AC_CHECK_FUNCS(posix_madvise)
```

Comment:
Both of these returned `yes` for e.g. the iOS arm build before your change, but now return `no`. I assume that platform doesn't like the `-Werror=strict-prototypes`.

Can you please move this to some SunOS-specific block?

Reply:
Does it have impact on the build? We can see that the build and relevant tests are succeeding on all platforms.

The problem was that `-Werror=strict-prototypes` is used during the compilation, but not during the introspection, so the introspection (passing) results were inconsistent and compilation was failing due to incomplete definition.

Reply:
We don't run the iOS tests as part of CI yet so hard to say.

In general it's pretty tricky to know for sure it doesn't have an impact and we were using the old behavior for years so I'd like to keep it :)

Reply:
Makes sense. Thanks for trying it out. I will wrap the `CFLAGS` set/unset in a condition.

Also just noticed that on the usage side, we don't have something like `#error "Platform not supported"` in the `#else` case (which is a typical pattern in coreclr), it silently gets compiled (maybe to accommodate non-POSIX variety of platforms).